### PR TITLE
#2094 Handle returning Value from execute_atom method in Python

### PR DESCRIPTION
**Pull request by stellarspot**
* Handle returning Value from execute_atom method (#2094)
* Unit test is added

### CODE REVIEW

File: opencog/cython/opencog/bindlink.pyx [hunk 1]
```diff
@@ -1,15 +1,21 @@
...
+from opencog.atomspace import is_a, types
+
 def execute_atom(AtomSpace atomspace, Atom atom):
     if atom == None: raise ValueError("execute_atom atom is: None")
     cdef cValuePtr c_value_ptr = c_execute_atom(atomspace.atomspace,
                                            deref(atom.handle))
-    cdef cHandle c_result = handle_cast(c_value_ptr)
-    return Atom.createAtom(c_result, atomspace)
+
+    if is_a(deref(c_value_ptr).get_type(), types.Atom):
```
Comment (vsbogd) [inline opencog/cython/opencog/bindlink.pyx]: Such implementation will not return correct class of Atom or Value. Please use `create_value_by_type` instead, see:
https://github.com/opencog/atomspace/blob/5b47b937b7cb1d03785042bf636ae0e2e90a5444/opencog/cython/opencog/atom.pyx#L71-L78

It handles both Atom and Value cases.

Comment (stellarspot) [reply opencog/cython/opencog/bindlink.pyx]: I updated the code to use create_value_by_type and updated the test to check that value.to_list()  works as expected.

File: opencog/cython/opencog/bindlink.pyx [hunk 2]
```diff
@@ -1,15 +1,22 @@
...
 
+from opencog.atomspace import is_a, types, create_value_by_type
+
 def execute_atom(AtomSpace atomspace, Atom atom):
     if atom == None: raise ValueError("execute_atom atom is: None")
     cdef cValuePtr c_value_ptr = c_execute_atom(atomspace.atomspace,
                                            deref(atom.handle))
-    cdef cHandle c_result = handle_cast(c_value_ptr)
-    return Atom.createAtom(c_result, atomspace)
+    type = deref(c_value_ptr).get_type()
```
Comment (vsbogd) [inline opencog/cython/opencog/bindlink.pyx]: Following three lines are not needed, `create_value_by_type` will construct `Atom` instance if required

Comment (stellarspot) [reply opencog/cython/opencog/bindlink.pyx]: Updated.

**Pull request merged**